src/puppetlabs/puppetdb/query_eng/engine.clj

```diff
@@ -19,18 +19,72 @@
             [puppetlabs.puppetdb.zip :as zip]
             [schema.core :as s])
   (:import [honeysql.types SqlCall SqlRaw]))
+
 ;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
 ;;; Plan - functions/transformations of the internal query plan
 
-(defrecord Query [projections selection source-table alias where
-                  subquery? entity call group-by limit offset order-by])
-(defrecord BinaryExpression [operator column value])
-(defrecord InArrayExpression [table column value])
-(defrecord RegexExpression [column value])
-(defrecord ArrayRegexExpression [column value])
-(defrecord NullExpression [column null?])
-(defrecord ArrayBinaryExpression [column value])
-(defrecord InExpression [column subquery])
+(def table-names #{"catalog_resources"
+                   "certnames"
+                   "edges"
+                   "environments"
+                   "fact_paths"
+                   "fact_values"
+                   "facts"
+                   "factsets"
+                   "latest_report"
+                   "reports"
+                   "resource_events"
+                   "resource_params"})
```

Comment:
OK - I wondered about that, but figured it should be pretty obvious where you needed to add the string, and this made it very clear what Query expects.  However, I hadn't thought about PE.  If we like, I can remove the set, though I might want to at least add  a comment making it clear that source-table should be a string table name.